docs-site/docs/dist-train.md

```diff
@@ -145,30 +165,80 @@ if __name__ == '__main__':
     nccl_id = singa.NcclIdHolder()
 
     # Define the number of GPUs to be used in the training process
-    gpu_per_node = int(sys.argv[1])
-    gpu_num = 1
+    num_gpus = int(sys.argv[1])
 
     # Define and launch the multi-processing
 	import multiprocessing
     process = []
-    for gpu_num in range(0, gpu_per_node):
+    for gpu_num in range(0, num_gpus):
         process.append(multiprocessing.Process(target=train_mnist_cnn,
-                       args=(nccl_id, gpu_num, gpu_per_node)))
+                       args=(nccl_id, gpu_num, num_gpus)))
 
     for p in process:
         p.start()
 ```
 
+Here are some explanations concerning the variables created above:
+
+(i) `nccl_id`
+
+Note that we need to generate a NCCL ID here to be used for collective communication, and then pass it to all the processes. 
+The NCCL ID is like a ticket, where only the processes with this ID can join the AllReduce operation. 
+(Later if we use MPI, the passing of NCCL ID is not necessary, because the ID is broadcased by MPI in our code automatically)
+
+(ii) `num_gpus`
```

Comment:
shall we simplify the terminology by using local_rank and global_rank?

Reply:
yes, can use <local_rank, global_rank, world size> in every locations of python and C++?